RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromTTracksEDProducer.cc

```diff
@@ -0,0 +1,169 @@
+    if (!tsosOnLayer.isValid()) {
+      std::cout << "ERROR!: Hitless TSOS is not valid! \n";
+    } else {
+      dets.front().second.rescaleError(errorSFHitless);
+
+      PTrajectoryStateOnDet const& ptsod =
+          trajectoryStateTransform::persistentState(tsosOnLayer, detOnLayer->geographicalId().rawId());
+      TrajectorySeed::RecHitContainer rHC;
+      if (numSeedsMade < 1) {  // only outermost seed
+        out->push_back(TrajectorySeed(ptsod, rHC, oppositeToMomentum));
+        numSeedsMade++;
+      }
+    }
+  }
+}
+
+void SeedGeneratorFromTTracksEDProducer::produce(edm::Event& ev, const edm::EventSetup& es) {
+  std::unique_ptr<std::vector<TrajectorySeed>> result(new std::vector<TrajectorySeed>());
+
+  // TTrack Collection
+  Handle<std::vector<TTTrack<Ref_Phase2TrackerDigi_>>> trks;
+  ev.getByToken(theInputCollectionTag, trks);
+
+  // Trk Geometry
+  edm::ESHandle<TrackerGeometry> tmpTkGeometryH;
+  es.get<TrackerDigiGeometryRecord>().get(tmpTkGeometryH);
+
+  // Mag field
+  edm::ESHandle<MagneticField> magfieldH;
+  es.get<IdealMagneticFieldRecord>().get(magfieldH);
+
+  // Estimator
+  edm::ESHandle<Chi2MeasurementEstimatorBase> estimatorH;
+  es.get<TrackingComponentsRecord>().get(theEstimatorName, estimatorH);
+
+  // Get Propagators
+  edm::ESHandle<Propagator> propagatorAlongH;
+  es.get<TrackingComponentsRecord>().get(thePropagatorName, propagatorAlongH);
+  std::unique_ptr<Propagator> propagatorAlong = SetPropagationDirection(*propagatorAlongH, alongMomentum);
+
+  edm::ESHandle<Propagator> propagatorOppositeH;
+  es.get<TrackingComponentsRecord>().get(thePropagatorName, propagatorOppositeH);
+  std::unique_ptr<Propagator> propagatorOpposite = SetPropagationDirection(*propagatorOppositeH, oppositeToMomentum);
+
+  // Get vector of Detector layers
+  edm::Handle<MeasurementTrackerEvent> measurementTrackerH;
+  ev.getByToken(theMeasurementTrackerTag, measurementTrackerH);
+  std::vector<BarrelDetLayer const*> const& tob = measurementTrackerH->geometricSearchTracker()->tobLayers();
+  std::vector<ForwardDetLayer const*> const& tecPositive =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->posTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->posTecLayers();
+  std::vector<ForwardDetLayer const*> const& tecNegative =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->negTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->negTecLayers();
+
+  /// Surface used to make a TSOS at the PCA to the beamline
+  Plane::PlanePointer dummyPlane = Plane::build(Plane::PositionType(), Plane::RotationType());
+
+  // Loop over the L1's and make seeds for all of them:
+  std::vector<TTTrack<Ref_Phase2TrackerDigi_>>::const_iterator it;
+  for (it = trks->begin(); it != trks->end(); it++) {
+    const TTTrack<Ref_Phase2TrackerDigi_>& l1 = (*it);
```

Comment:
@slava77 Not sure this will work, I get the following error if I switch to auto:
```
error: no matching function for call to 'begin(edm::Handle<std::vector<TTTrack<edm::Ref<edm::DetSetVector<Phase2TrackerDigi>, Phase2TrackerDigi> > > >&)'
```

Reply:
this implies https://github.com/cms-sw/cmssw/pull/30342#discussion_r447798921 is applied 
```  auto const& trks = ev.get(theInputCollectionTag);```